src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1008,6 +1024,10 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMProposeSet> const& m)
 {
     protocol::TMProposeSet& set = *m;
 
+    if (overlay_.setup().expire &&
+            set.has_hops() && ! slot_->cluster())
+        set.set_hops(set.hops() + 1);
+
```

Comment:
Nope, can't be changed. The C++ code for protocol buffers is generated by the *protoc* tool.